### PR TITLE
Make lowercase columns default, add option to make them same as model

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR Addresses https://github.com/ODM2/ODM2PythonAPI/issues/127#issuecomment-356519409. Now `getResultValues` and `getDatasetsValues` have an additional argument called `lowercols`, which defaults to `True` to keep the api the same, and make changes less disruptive. I have tested this and it works.

I was wondering if I should create some sort of warning here too saying how this argument will be deprecated in the next release after this one, and the `*values` dataframe will default to columns similar to the model in future releases, therefore user have to add `lowercase=False` to anticipate for that?

### CODE REVIEW

**Comment by emiliom** [comment]
> I was wondering if I should create some sort of warning here too saying how this argument will be deprecated in the next release after this one, and the *values dataframe will default to columns similar to the model in future releases, therefore user have to add lowercase=False to anticipate for that?

YES! That'd be perfect. Though at this time we don't need to be specific about which release it'll be deprecated on. We can say "soon" or "in the near future", then once a decision is made about a timeframe, we can change the warning to be more specific.

Regarding `getDatasetsValues `: I didn't realize that function was also impacted by this issue/decision. But I guess I hadn't really thought about or examined that function. Thanks for making the two functions consistent.

**Comment by lsetiawan** [comment]
Is this warning message okay?

```python
'The parameter \'lowercols\' default makes the column names lowercase. '
 'Please set this to False. \'lowercols\' will be deprecated in the near future; '
 'column names casing will then be similar to its respective model.'
```

**Comment by emiliom** [comment]
@lsetiawan See the conflicts that were identified by github. I'll hold off on reviewing this PR until the conflict is resolved. Thanks.

File: odm2api/ODM2/services/readService.py [hunk 1]
```diff
@@ -933,6 +933,8 @@ def getDataSetsValues(self, ids=None, codes=None, uuids=None, dstype=None):
             uuids (list, optional): List of Dataset UUIDs string.
             dstype (str, optional): Type of Dataset from
                 `controlled vocabulary name <http://vocabulary.odm2.org/datasettype/>`_.
+            lowercols (bool, optional): Make column names to be lowercase.
+                                        Default to True.
```
Comment (emiliom) [inline odm2api/ODM2/services/readService.py]: Add this at the end of the `lowercols` description: "Please start upgrading your code to rely on CamelCase column names. In a near-future release, the default will be changed to False, and later the parameter may be removed.

File: odm2api/ODM2/services/readService.py [hunk 2]
```diff
@@ -1347,6 +1349,8 @@ def getResultValues(self, resultids, starttime=None, endtime=None):
             resultids (list): List of SamplingFeatureIDs.
             starttime (object, optional): Start time to filter by as datetime object.
             endtime (object, optional): End time to filter by as datetime object.
+            lowercols (bool, optional): Make column names to be lowercase.
+                                        Default to True.
```
Comment (emiliom) [inline odm2api/ODM2/services/readService.py]: Add this at the end of the `lowercols` description: "Please start upgrading your code to rely on CamelCase column names. In a near-future release, the default will be changed to False, and later the parameter may be removed.

**Comment by emiliom** [comment]
See my suggested changes to a couple of doc strings. If you agree with them, go ahead and merge this PR yourself once you've made that addition. Thanks!

**Comment by lsetiawan** [comment]
@ocefpaf For some reason, Travis CI is not starting, I've been waiting for it for few hours, I restarted it after about an hour. Do you know what's going on? Thanks!

**Comment by ocefpaf** [comment]
@lsetiawan take a look at their status bot on Twitter. It seems that they had an incident a while ago:

https://twitter.com/traviscistatus

**Comment by lsetiawan** [comment]
Oh okay. Good to know. I'll just let it do it's thing then. Thanks!

**Comment by lsetiawan** [comment]
Travis finally worked. Merging.

**Pull request merged**